Lib/concurrent/futures/_base.py

```diff
@@ -648,6 +695,25 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         return False
 
 
+class _MapResultIterator(object):
```

Comment:
`(object)` is redundant.

Current iterator has the `close()` method, calling which cancels all futures. It is a useful feature. Please add a `close()` method in a new class. I do not think that it is worth to implement `send()` and `throw()`, they are not so useful and are rather an implementation detail.